### PR TITLE
Add the current-routes resource

### LINKED ISSUE #73: New endpoint for skipper /current-routes
Skipper doesn't care about all of the routes, but only about the current ones. Create an endpoint where skipper can get the current state of the routes.

The `/current-routes` resource, has the route name as an `id`.

### PR DESCRIPTION
**Pull request by danpersa**
Closes #73 

@mo-gr @aryszka @c00ler can you take a look?

### CODE REVIEW

File: src/it/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepoSpec.scala [hunk 1]
```diff
@@ -88,9 +77,9 @@ class RoutesPostgresRepoSpec extends FunSpec with BeforeAndAfter with Matchers w
 
           it("should select all routes") {
             val createdAt = LocalDateTime.now()
-            val activateAt = createdAt.plusMinutes(5)
-            insertRoute("R1", "/hello1", createdAt = createdAt)
-            insertRoute("R2", "/hello2", createdAt = createdAt)
+            val activateAt = createdAt.minusMinutes(5)
+            insertRoute("R1", "/hello1", createdAt = createdAt, activateAt = activateAt)
```
Comment (mo-gr) [inline src/it/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepoSpec.scala]: activated before created?

File: src/it/scala/org/zalando/spearheads/innkeeper/routes/DeleteRouteSpec.scala [hunk 2]
```diff
@@ -64,7 +64,7 @@ class DeleteRouteSpec extends FunSpec with BeforeAndAfter with Matchers {
             insertRoute("R2", ownedByTeam = "team1")
 
             val response = deleteSlashRoute(2, token)
-            response.status.shouldBe(StatusCodes.OK)
+            response.status should be(StatusCodes.OK)
```
Comment (mo-gr) [inline src/it/scala/org/zalando/spearheads/innkeeper/routes/DeleteRouteSpec.scala]: syntax sugar overload... I actually prefer the previous version. But it is a matter of personal preference

File: src/it/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepoSpec.scala (hunk 1 above)
Comment (danpersa) [inline src/it/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepoSpec.scala]: be sure that the route is always active...

File: src/it/scala/org/zalando/spearheads/innkeeper/routes/DeleteRouteSpec.scala (hunk 2 above)
Comment (danpersa) [inline src/it/scala/org/zalando/spearheads/innkeeper/routes/DeleteRouteSpec.scala]: This is related to #68 make everything consistent...

File: src/it/scala/org/zalando/spearheads/innkeeper/routes/GetCurrentRoutesSpec.scala [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        insertRoute("R3", createdAt = createdAt, activateAt = activateAt)
+        deleteRoute(5)
+
+        val response = getSlashCurrentRoutes(token)
+
+        response.status should be(StatusCodes.OK)
+        val entity = entityString(response)
+        val routes = entity.parseJson.convertTo[Seq[RouteOut]]
+        //routes.size should be (3)
+        routes.map(_.id).toSet should be (Set(1, 4, 6))
```
Comment (mo-gr) [inline src/it/scala/org/zalando/spearheads/innkeeper/routes/GetCurrentRoutesSpec.scala]: is it possible to go assert against the route name here? Like `... should be (Set("R1", "R4", "R3"))`? I think this would make the intention of the test a lot more clear. Right now, I don't really know where 1, 4 and 6 come from. I assume, they rely on the fact, that ids are auto generated and incremental. This is an implementation detail and make the test unnecessarily hard to follow

Comment (danpersa) [inline src/it/scala/org/zalando/spearheads/innkeeper/routes/GetCurrentRoutesSpec.scala]: The name is not unique... Bot routes with id 6 and 3 have the same name, `R3`... but we pick the route with id 6

File: src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala [hunk 4]
```diff
@@ -73,13 +73,8 @@ class RoutesPostgresRepo @Inject() (
...
     logger.debug("select all")
 
-    val q = for {
-      routeRow <- routesTable
-      if routeRow.deletedAt.isEmpty
-    } yield routeRow
-
     db.stream {
-      q.result
+      selectAllQuery.result
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala]: wow... This significantly changes the behaviour of the method (previously: return all non-deleted routes, now: return all routes). I hope, all code, that relied on the previous behaviour has been updated?

Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala]: Especially since other methods, like `selectByName` still omit deleted routes

Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala]: we still omit the deleted routes here also...

Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala]: I just extracted the query as a val...

File: src/it/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepoSpec.scala (hunk 1 above)
Comment (mo-gr) [inline src/it/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepoSpec.scala]: still... sounds weird. How can I activate something before it is created? shouldn't it be the other way around? activated is `now - epsilon` and created is `now - 2epsilon`?

But this seems to be common thing through this PR. If it works... keep it. I just think it is a bit confusing to readers

File: src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala [hunk 5]
```diff
@@ -99,16 +94,38 @@ class RoutesPostgresRepo @Inject() (
...
       q.result
     }
   }
 
+  override def selectLatestActiveRoutesPerName(currentTime: LocalDateTime): DatabasePublisher[RouteRow] = {
+    logger.debug("select latest routes per name")
+
+    val q = (for {
+      routeRow <- routesTable
+      if routeRow.deletedAt.isEmpty && routeRow.activateAt < currentTime
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala]: Is it possible to deactivate a route? Or can I just `create -> activate -> delete` with no way back?

File: src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala [hunk 6]
```diff
@@ -99,16 +94,38 @@ class RoutesPostgresRepo @Inject() (
...
+    } yield (routeRow.id, routeRow.name)).groupBy(_._2)
+
+    val maxIdsQuery = q.map {
+      case (name, query) =>
```
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala]: why is this named `query`? It isn't a query, is it?

File: src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala (hunk 4 above)
Comment (mo-gr) [inline src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala]: ok. forget everything I just wrote... Found the private lazy val at the bottom of the file... never mind. carry on.

File: src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala (hunk 5 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala]: routes are immutable... when you create a route, you can specify an `activateAt` time, which can't be changed... if you want a route not to be valid anymore, you have to delete it...

File: src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala (hunk 6 above)
Comment (danpersa) [inline src/main/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepo.scala]: for some reason, its type actually is: `Query[(Rep[Long], Rep[String]), (Long, String), Seq]`

File: src/test/scala/org/zalando/spearheads/innkeeper/services/RoutesServiceSpec.scala [hunk 7]
```diff
@@ -26,13 +27,14 @@ import scala.language.postfixOps
  */
 class RoutesServiceSpec extends FunSpec with Matchers with MockFactory with ScalaFutures {
 
+  override implicit val patienceConfig = PatienceConfig(timeout = Span(1000, Seconds))
```
Comment (mo-gr) [inline src/test/scala/org/zalando/spearheads/innkeeper/services/RoutesServiceSpec.scala]: 1000 seconds? Isn't that a bit too patient? If a test runs > 15 min, something is probably broken

File: src/it/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepoSpec.scala (hunk 1 above)
Comment (danpersa) [inline src/it/scala/org/zalando/spearheads/innkeeper/dao/RoutesPostgresRepoSpec.scala]: You can open a ticket to improve readability :+1:

**Comment by mo-gr** [comment]
lgtm

File: src/test/scala/org/zalando/spearheads/innkeeper/services/RoutesServiceSpec.scala (hunk 7 above)
Comment (danpersa) [inline src/test/scala/org/zalando/spearheads/innkeeper/services/RoutesServiceSpec.scala]: I forgot this line here... I'll remove it...
